### DISCUSSION TITLE
添加了一个关于个人微信掉线的邮件提醒

### DISCUSSION THREAD

**Discussion by dentes9988** [zhayujie/chatgpt-on-wechat#1601] [Ideas]
用gpt去写的代码，真的好玩

import time
import smtplib
from email.mime.text import MIMEText
from email.header import Header

def send_email():
    sender = 'your_email@example.com'  # 替换为您的发送邮箱
    receivers = 'recipient_email@example.com'  # 替换为接收者的邮箱
    message = MIMEText('AI机器人已掉线', 'plain', 'utf-8')
    message['From'] = sender
    message['To'] = receivers
    subject = 'AI机器人已掉线'
    message['Subject'] = Header(subject, 'utf-8')

    try:
        smtpObj = smtplib.SMTP('smtp.example.com', 25)  # 替换为您的SMTP服务器和端口
        smtpObj.login('your_email@example.com', 'your_password')  # 替换为您的邮箱和授权码
        smtpObj.sendmail(sender, [receivers], message.as_string())
        smtpObj.quit()
        print("邮件发送成功")
    except smtplib.SMTPException as e:
        print("邮件发送失败:", e)

def check_log(last_position):
    with open('/path/to/your/logfile.log', 'r') as file:  # 替换为您的日志文件路径
        file.seek(last_position)
        content = file.read()
        new_position = file.tell()

    if 'LOG OUT!' in content:
        send_email()
    
    return new_position

last_position = 0
last_position = check_log(last_position)  # 执行初始检测

while True:
    last_position = check_log(last_position)
    time.sleep(180)  # 每隔3分钟执行一次

![image](https://github.com/zhayujie/chatgpt-on-wechat/assets/51904662/f0e80b92-6b93-43f3-a7d8-adaf8b2d6c71)

**Comment by babyowen** [comment]
这个python脚本能保持后台一直运行吗，会不会也被踢出去

**Reply by dentes9988** [reply]
可以参考服务器部署那块的介绍，在后台运行，这个应该不会踢出去
<img width="862" alt="image" src="https://github.com/zhayujie/chatgpt-on-wechat/assets/51904662/b05caea3-1387-4b6f-910d-988730a26829">
